gammapy/cube/core.py

```diff
@@ -448,10 +448,12 @@ def integral_flux_image(self, energy_band, energy_bins=10):
         axes = ['longitude', 'latitude']
         header = self.wcs.sub(axes).to_header()
 
-        hdu = fits.ImageHDU(data=integral_flux,
-                            header=header, name='integral_flux')
-
-        return SkyMap.from_image_hdu(hdu)
+        image = SkyMap(name='flux',
+                       data=integral_flux,
+                       wcs=self.wcs.sub(axes),
```

Comment:
At this moment `wcs` has only two axes so this sub does nothing, but if something changes I think it will be more logical to keep parameter in that way.